src/main/java/spoon/support/QueueProcessingManager.java

```diff
@@ -105,26 +106,34 @@ protected ProcessingVisitor getVisitor() {
 	public void process(Collection<? extends CtElement> elements) {
 		Processor<?> p;
 		while ((p = getProcessors().poll()) != null) {
-			getFactory().getEnvironment().reportProgressMessage(p.getClass().getName());
-			current = p;
-			p.initProperties(AbstractProcessor.loadProperties(p));
-			p.init();
-			p.process();
-			for (CtElement e : new ArrayList<CtElement>(elements)) {
-				process(e, p);
+			try {
+				getFactory().getEnvironment().reportProgressMessage(p.getClass().getName());
+				current = p;
+				p.initProperties(AbstractProcessor.loadProperties(p));
+				p.init();
+				p.process();
+				for (CtElement e : new ArrayList<CtElement>(elements)) {
+					process(e, p);
+				}
+				p.processingDone();
```

Comment:
If `ProcessInterrupt` is thrown in  `processingDone` the method `processingDone` will be called two times.